### ISSUE TITLE
Upgrading to 2.2.x results in type errors when importing from werkzeug.routing

### ISSUE THREAD

**Issue by alisaifee** [pallets/werkzeug#2492]
After upgrading to werkzeug 2.2.1 importing any class from `werkzeug.routing` results in an error from mypy if `no_implicit_reexport=True`. This was not the case in previous versions as `werkzeug.routing` was a single file submodule. 

### Reproduction
Given `eg.py`:
```python
from werkzeug.routing import Rule
```
With `werkzeug==2.2.1`
```shell
$ mypy eg.py --strict
eg.py:1: error: Module "werkzeug.routing" does not explicitly export attribute "Rule"; implicit reexport disabled  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

With `werkzeug==2.1.0`
```shell
$ mypy eg.py --strict
Success: no issues found in 1 source file```
```

### Environment:

- Python version: 3.10
- Werkzeug version: 2.2.1

**Comment by mrphlip** [comment]
There are other objects, too, which are now just unavailable to be imported from this module... eg `from werkzeug.routing import NotFound` gives an `ImportError`.

Yes, you can just import it from `werkzeug.exceptions` instead, but existing code in other libraries (and the [current documentation](https://werkzeug.palletsprojects.com/en/2.2.x/routing/#quickstart)) import it from here.

**Comment by davidism** [comment]
Those have been fixed. If one is not fixed in the current point release, make a separate report about that.

**Comment by davidism** [comment]
We'll fix the current documentation, but please fix the imports in the other libraries. NotFound should only be exported by exceptions.

**Comment by alisaifee** [comment]
The specific reported issue of attributes that are expected to be exported by `werkzeug.routing` is resolved by #2493 so closing this.
